docs/usage/key-concepts/how-it-works.md

```diff
@@ -0,0 +1,152 @@
+---
+title: How it works ?
+description: Learn all about Renovate's workflow
+---
+
+# Introduction
+
+Renovate works by searching for all the dependencies in your project and then look for updates for each.
+
+To be compatible with the variety of dependency-naming and versioning conventions, Renovate has defined modules for each known convention, and you can define your own too.
+
+Please add comments to the [issue#25091](https://github.com/renovatebot/renovate/issues/25091) if you wish to see a part (better) shown in the graph below
+
+## Modules
+
+The three modules are: [manager](../modules/manager/index.md), [datasource](../modules/datasource/index.md) and [versioning](../modules/versioning.md).
+
+They are used sequentially:
+
+1. the manager module looks for files based on their name and extract dependencies from them, each dependency has a datasource
+2. the datasource module looks for the existing versions of the dependency
+3. the versioning module search for a valid version regarding the dependency's version
```

Comment:
I don't understand this sentence. I think you mean something like:

> The versioning modules finds valid versions, based on defined rules

As far as I understand, the versioning modules has rules about what's a valid version. So the versioning module only matches known valid versions?

Reply:
By "valid" I meant, the one matching the rules : the higher minor, or the highest patch, of the one keeping the compatiblity suffix `-alpine`, ...

The datasource will give all existing versions, and versioning will select the good one regarding the rules